src/android/com/mady/wifi/api/WifiHotSpots.java

```diff
@@ -247,6 +250,33 @@ public boolean connectToHotspot(String netSSID, String netPass, Integer authenti
         return false;
     }
 
+    /**
+     * Determines if current connection is a captive portal by checking if we get a 204 response
+     * from google. If we do not, we can assume this is do to a captive portal instead returning a 200 response.
+     * Logic taken from https://stackoverflow.com/questions/13958614/how-to-check-for-unrestricted-internet-access-captive-portal-detection
+     */
+    public boolean isCaptivePortalConnection() {
+        HttpURLConnection urlConnection = null;
+        try {
+            URL url = new URL("http://clients3.google.com/generate_204");
+            urlConnection = (HttpURLConnection) url.openConnection();
+            urlConnection.setInstanceFollowRedirects(false);
+            urlConnection.setConnectTimeout(10000);
+            urlConnection.setReadTimeout(10000);
+            urlConnection.setUseCaches(false);
+            urlConnection.getInputStream();
+            // We got a valid response, but not from the real google
+            return urlConnection.getResponseCode() != 204;
+        } catch (IOException e) {
+            Log.d("RAVEN", "Captive portal check - probably not a portal: exception " + e.toString());
+            return false;
```

Comment:
If there is an internet connection problem, it is resolving to not be a captive portal. Is that a right condition, I think in that case you should throw an error back to UI.

Looking at the code below, you can throw the Exception which can be caught below.

Reply:
I think it is handling that condition right, we would not want this function to throw an additional error if the user does not have a valid connection. We only use it to determine if the connection has a captive portal. If it is not a valid connection, we can not determine if it has a captive portal so we return false.

Also, in theory the workflow should never get to the `isCaptivePortalConnection` check if it wasn't a valid connection.